assets/js/modules/adsense/components/setup/v2/SetupAccount.js

```diff
@@ -78,22 +78,22 @@ export default function SetupAccount( { account } ) {
 
 	useEffect( () => {
 		// Do nothing if clients aren't loaded because we can't determine acfClientID yet.
-		if ( clients === undefined ) {
+		if ( clients === undefined || site === undefined ) {
 			return;
 		}
 
 		if ( ! acfClientID ) {
 			setAccountStatus( ACCOUNT_STATUS_NO_CLIENT );
-		} else if ( state === SITE_STATE_NEEDS_ATTENTION ) {
+		} else if ( accountState === STATE_NEEDS_ATTENTION ) {
 			setAccountStatus( ACCOUNT_STATUS_NEEDS_ATTENTION );
-		} else if ( state === SITE_STATE_REQUIRES_REVIEW ) {
+		} else if ( site?.state === STATE_REQUIRES_REVIEW ) {
 			setAccountStatus( ACCOUNT_STATUS_CLIENT_REQUIRES_REVIEW );
-		} else if ( state === SITE_STATE_GETTING_READY ) {
+		} else if ( site?.state === STATE_GETTING_READY ) {
 			setAccountStatus( ACCOUNT_STATUS_CLIENT_GETTING_READY );
```

Comment:
This is still not correct. Please review https://github.com/google/site-kit-wp/issues/5052#issuecomment-1101907230 and https://github.com/google/site-kit-wp/pull/5083#discussion_r852525037.

You're now looking at the _site's_ state. But as mentioned in my previous comments, you need to look at the _client's_ state. The client is different from the account and also different from the site. We need to look at the AFC client in the account and use _that_ state.

There is no dedicated store selector to get that client yet, we only have `getClients( accountID )`. So we may want to add something like `getAFCClient( accountID )` that calls `getClients` and parses out the AFC client (or returns `null` if there is none).